module_evaluations/TCR-6_2022-09-28.MD

```diff
@@ -0,0 +1,60 @@
+# Module acceptance criteria template
+
+## How to use this form
+When performing a technical evaluation of a module, create a copy of this document and use the conventions below to indicate the status of each criteria.  The evaluation results should be placed in the [module_evaluations](https://github.com/folio-org/tech-council/tree/master/module_evaluations) directory and should conform to the following naming convention:  `{JIRA Key}_YYYY-MM-DD.MD`, e.g. `TCR-1_2021-11-17.MD`.  The date here is used to differenciate between initial and potential re-evaluation(s).  It should be the date which the evaluation results file was created.
+
+* [x] ACCEPTABLE
+* [x] ~INAPPLICABLE~
+* [ ] UNACCEPTABLE
+  * comments on what was evaluated/not evaluated, why a criterion failed
+
+## [Criteria](https://github.com/folio-org/tech-council/blob/4dc5ac6d632fb880131cd214d212f973683b9ebe/MODULE_ACCEPTANCE_CRITERIA.MD)
+
+* [X] Upon acceptance, code author(s) agree to have source code canonically in folio-org github
+* [X] Copyright assigned to OLF
+* [X] Uses Apache 2.0 license
+* [X] Third party dependencies use an Apache 2.0 compatible license
+* [X] Module’s repository includes a compliant Module Descriptor
+* [X] Modules must declare all consumed interfaces in the Module Descriptor “requires” and “optional” sections
+* [ ] Environment vars are documented in the ModuleDescriptor - _note: read more at [https://wiki.folio.org/pages/viewpage.action?pageId=65110683](https://wiki.folio.org/pages/viewpage.action?pageId=65110683)_
+  * The following documented environment variables are not present in the module descriptor: `EVENTS_PUBLISHER_BOOTSTRAP_SERVERS`, `EVENTS_CONSUMER_BOOTSTRAP_SERVERS`, `EVENTS_PUBLISHER_ZK_CONNECT`, `EVENTS_CONSUMER_ZK_CONNECT`
+* [X] Back-end modules must define endpoints consumable by other modules in the Module Descriptor “provides” section
+* [ ] All API endpoints are documented in RAML or OpenAPI
+  * This work is being done by the developers, but it is not yet complete.
+* [X] All API endpoints protected with appropriate permissions
+* [X] No excessive permissions granted to the module
+* [X] Code of Conduct statement in repository
+* [ ] Installation documentation included
+  * Documentation which assits an implementor in getting this module runnin locally would satisfy this requirement 
+* [X] Contribution guide is included in repo
+* [X] Module provides reference data (if applicable)
+* [X] Personal data form is completed, accurate, and provided as PERSONAL_DATA_DISCLOSURE.md file
+* [X] Sensitive information is not checked into git repository
+* [X] Module is written in a language and framework that FOLIO development teams are familiar with _e.g. Vertx/RMB, Spring Way/folio-spring-base, and React/Stripes_
+  * Groovy is already in use in other FOLIO modules
```

Comment:
> "Groovy is already in use"

@jeremythuff, I don't think this is relevant. The [acceptance criteria](https://github.com/folio-org/tech-council/blob/624c01b8092270c68f2d4ca55dbc2364a6da38f0/MODULE_ACCEPTANCE_CRITERIA.MD) state "Module is written in a language and framework from the [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) page". Groovy is not listed there for any past or future release.

Reply:
> Groovy is not listed there for any past (Lotus, Morning Glory) or future (Nolana, Orchid) release.

I'd like to check my understanding of this stance.

My interpretation of this is that this module should not be accepted by the TC because groovy is not in the supported technology list. And (by inference, from the reference to past releases) that historical (pre-approval) modules that don't use supported technologies should also not be part of the flower releases.

How wrong is my interpretation of this stance?

Reply:
> My interpretation of this is that this module should not be accepted by the TC because groovy is not in the supported technology list. 

Yes, that is my interpretation too, though I am not on the TC. 

> And (by inference, from the reference to past releases) that historical (pre-approval) modules that don't use supported technologies should also not be part of the flower releases.

No, I strongly disagree with this conclusion. I did not mean to imply anything about "historical (pre-approval) modules". It is my understanding that these criteria apply to new modules only, hence "New Module Technical Evaluation Process".

Reply:
> No, I strongly disagree with this conclusion. I did not mean to imply anything about "historical (pre-approval) modules". It is my understanding that these criteria apply to new modules only, hence "New Module Technical Evaluation Process".

I extrapolated once I connected the dots together between the criteria, the supported technology list and modules.

I think we've expanded the context of this by defining the supported technology list for flower releases. The upcoming process conversations in that area include topics about how to synchronise infrastructure changes across existing modules. Thus to me, the reach of that document is heading towards all modules.

We've also been talking about wanting to apply the criteria to existing modules.

I was expecting us (the TC et al) not to have agreement on this aspect.

Reply:
Apologies for not making it more obvious the distinction between my interpretation of what you wrote and my thinking about where I believe that might lead.